tools/docker/scripts/ci_entrypoint.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+# author: Ole Schuett
+
+REPORT=/workspace/report.txt
+echo -n "StartDate: " > $REPORT
+date --utc --rfc-3339=seconds >> $REPORT
+
+# Get cp2k sources.
+if [ -n "${GIT_REF}" ]; then
+    echo -e "\n========== Fetching Git Commit =========="
+    cd /workspace/cp2k
+    git fetch origin "${GIT_BRANCH}"
+    git checkout "${GIT_REF}"
+    git submodule update --init --recursive
+    git --no-pager log -1 --pretty='%nCommitSHA: %H%nCommitTime: %ci%nCommitAuthor: %an%nCommitSubject: %s%n' | tee -a $REPORT
+
+elif [ -d  /mnt/cp2k ]; then
+    echo -e "\n========== Copying Changed Files =========="
+    #TODO: Exclusion rules could also come from .gitigore
+    rsync --exclude="*~"                      \
```

Comment:
you might want to either `rm -rf /workspace/cp2k/` or use `--delete` to make sure to clear removed files, especially with a build system which picks up files automatically

Reply:
Good catch - thanks!